### PR TITLE
lint: add switch -> if suggestions for 1-case statements

### PR DESCRIPTION
**Pull request by weeellz**
Works for all kind of switch.

### CODE REVIEW

**Comment by weeellz** [comment]
@Quasilyte, should checker suggest alternative?

**Comment by Quasilyte** [comment]
@weeellz, I think we will have a good documentation one day that will explain how to address every warning and what alternatives are implied. Maybe we can suggest alternatives with special kfulint flag or with other command.
In any case, it's out of scope of our current tasks.

I'll do a review tomorrow if you don't mind.

**Comment by weeellz** [comment]
@Quasilyte, ok.

File: lint/switchif_checker.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	ctx *Context
+}
+
+func newSwitchifChecker(ctx *Context) Checker {
+	return &switchifChecker{ctx: ctx}
+}
+
+func (c *switchifChecker) Check(f *ast.File) {
+	ast.Inspect(f, func(x ast.Node) bool {
+		if stmt, ok := x.(*ast.SwitchStmt); ok {
```
Comment (fexolm) [inline lint/switchif_checker.go]: You could use typeswitch here

Comment (fexolm) [reply lint/switchif_checker.go]:
``` 
switch stmt.(type) {
   case *ast.SwitchStmt:
       c.checkSwitchStmt(stmt, stmt.Body)
   ...
}

**Pull request merged**